src/coord/src/coord.rs

```diff
@@ -2138,14 +2170,50 @@ impl Coordinator {
 
     async fn sequence_end_transaction(
         &mut self,
-        session: &mut Session,
+        tx: ClientTransmitter<ExecuteResponse>,
+        mut session: Session,
         action: EndTransactionAction,
-    ) -> Result<ExecuteResponse, CoordError> {
+    ) {
         let was_implicit = matches!(
             session.transaction(),
             TransactionStatus::InTransactionImplicit(_)
         );
+        let rx = self
+            .sequence_end_transaction_inner(&mut session, &action)
+            .await;
+        match rx {
+            Ok(Some(rx)) => {
+                tokio::spawn(async move {
+                    let tag = match rx.await {
+                        Ok(Ok(_)) => action.tag(),
+                        // TODO: Try to return these errors somehow.
+                        Ok(Err(_err)) => EndTransactionAction::Rollback.tag(),
+                        // This case means the persistence runtime is no longer
+                        // running (shut down or crashed)s
+                        Err(_) => EndTransactionAction::Rollback.tag(),
+                    };
+                    let result = Ok(ExecuteResponse::TransactionExited { tag, was_implicit });
+                    tx.send(result, session);
+                });
+            }
+            Ok(None) => {
+                let result = Ok(ExecuteResponse::TransactionExited {
+                    tag: action.tag(),
+                    was_implicit,
+                });
+                tx.send(result, session);
+            }
+            Err(err) => {
+                tx.send(Err(err), session);
+            }
+        }
+    }
 
+    async fn sequence_end_transaction_inner(
```

Comment:
#7593 got merged, so this can be made not async now.

Reply:
👍 Working through the rebase now and I've made it not async